modules/auxiliary/admin/dcerpc/cve_2022_26923_certifried.rb

```diff
@@ -0,0 +1,513 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::SMB::Client::Authenticated
+  alias connect_smb_client connect
+
+  include Msf::Exploit::Remote::Kerberos::Client
+
+  include Msf::Exploit::Remote::LDAP
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::MsIcpr
+  include Msf::Exploit::Remote::MsSamr
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Active Directory Certificate Services (ADCS) privilege escalation (Certifried)',
+        'Description' => %q{
+          This module exploits a privilege escalation vulnerability in Active
+          Directory Certificate Services (ADCS) to generate a valid certificate
+          impersonating the Domain Controller (DC) computer account. This
+          certificate is then used to authenticate to the target as the DC
+          account using PKINIT preauthentication mechanism. The module will get
+          and cache the Ticket-Granting-Ticket (TGT) for this account along
+          with its NTLM hash. Finally, it requests a TGS impersonating a
+          privileged user (Administrator by default). This TGS can then be used
+          by other modules or external tools.
+
+          This will go through the following steps:
+          1. Check if the current user `ms-DS-MachineAccountQuota` let him add a computer account
+          2. Create a computer account
+          3. Change the new computer's `dNSHostName` attribute to match that of the DC
+          4. Request a certificate for this computer account and cache it
+          5. Authenticate to the remote host with the DC account's certificate and cache the TGT
+          6. Retrieve the DC account's NTLM hash
+          7. Escalate privileges by requesting a TGS impersonating a privileged domain user
+          8. Delete the computer account (only possible if the privilege
+          escalation succeeded or if the current user is an administrator)
```

Comment:
This doesn't render correctly in the module's output in `msfconsole`. We should probably remove it and just keep it in the module docs.